### PR TITLE
realm tests moved to pytest

### PR DESCRIPTION
**Pull request by pondrejk**
```
pytest tests/foreman/cli/test_realm.py -k test_positive_delete_by_name
================================================ test session starts ================================================

tests/foreman/cli/test_realm.py .                                                                              [100%]

============================================= 1 passed in 28.95 seconds =============================================
```

### CODE REVIEW

File: tests/foreman/cli/test_realm.py [hunk 1]
```diff
@@ -290,5 +285,5 @@ def test_negative_info_nonexistent_realm_id(self):
 
         :expectedresults: Realm not found
         """
-        with self.assertRaises(CLIReturnCodeError):
+        with pytest.raises(CLIReturnCodeError):
             Realm.info({'id': 0})
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_realm.py:289]: Good Candidate, to merge the test case?

File: tests/foreman/cli/test_realm.py [hunk 2]
```diff
@@ -29,128 +29,119 @@
...
+    def test_positive_delete_by_name(self, _make_proxy):
         """Realm deletion by realm name
 
         :id: ef3967e6-d53d-4dec-b74f-c20448d5fc6d
 
         :expectedresults: Realm is deleted
         """
-        proxy = self._make_proxy()
-        realm = make_realm({'realm-proxy-id': proxy['id'], 'realm-type': 'Active Directory'})
+        # proxy = _make_proxy
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_realm.py:55]: I hope this comment is not required.

File: tests/foreman/cli/test_realm.py (hunk 1 above)
Comment (pondrejk) [reply tests/foreman/cli/test_realm.py:289]: I see what you mean but I suppose there is not much to gain here time-wise

**Comment by omkarkhatavkar** [comment]
We will need at least another reviewer or tier2 reviewer

File: tests/foreman/cli/test_realm.py [hunk 3]
```diff
@@ -29,128 +29,118 @@
...
+    def test_positive_delete_by_name(self, _make_proxy):
         """Realm deletion by realm name
 
         :id: ef3967e6-d53d-4dec-b74f-c20448d5fc6d
 
         :expectedresults: Realm is deleted
         """
-        proxy = self._make_proxy()
-        realm = make_realm({'realm-proxy-id': proxy['id'], 'realm-type': 'Active Directory'})
+        realm = make_realm({'realm-proxy-id': 1, 'realm-type': 'Active Directory'})
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_realm.py:55]: Is it good idea to hardcode proxy id like that? It used to be id of proxy created for test.

Comment (pondrejk) [reply tests/foreman/cli/test_realm.py]: My bad, a leftover from development

File: tests/foreman/cli/test_realm.py [hunk 4]
```diff
@@ -29,128 +29,118 @@
...
-    def test_positive_delete_by_name(self):
+    def test_positive_delete_by_name(self, _make_proxy):
         """Realm deletion by realm name
 
         :id: ef3967e6-d53d-4dec-b74f-c20448d5fc6d
 
         :expectedresults: Realm is deleted
         """
-        proxy = self._make_proxy()
         realm = make_realm({'realm-proxy-id': proxy['id'], 'realm-type': 'Active Directory'})
```
Comment (mirekdlugosz) [inline tests/foreman/cli/test_realm.py:55]:
```suggestion
        realm = make_realm({'realm-proxy-id': _make_proxy['id'], 'realm-type': 'Active Directory'})
```
?
flake8 on travis complains there's undefined variable `proxy` here.

**Comment by pondrejk** [comment]
@mirekdlugosz can this go in?

**Review by mirekdlugosz — APPROVED** [review]
@pondrejk Yes, I must have missed this specific notification from GitHub. Sorry!

ACK and merge

**Pull request merged**